catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java

```diff
@@ -406,7 +426,8 @@ public String getFieldSuffix(AttributeFormat format) {
     }
 
     public MetacardType getMetacardType(SolrDocument doc) throws MetacardCreationException {
-        String mTypeFieldName = doc.getFieldValue(SchemaFields.METACARD_TYPE_FIELD_NAME).toString();
+        String mTypeFieldName = doc.getFieldValues(SchemaFields.METACARD_TYPE_FIELD_NAME)
+                .toArray()[0].toString();
```

Comment:
Silly me, I realize now I was looking in the wrong interface for exactly that. I'll fix them all.